### PR TITLE
NodeJS Course Getting Started: Add a reading assignment for createserver

### LINKED ISSUE #26558: NodeJS/Getting Started: Add http.createServer documentation to HTTP section
### Describe your suggestion

createServer is an essential component of the upcoming project (I think), however this block of documentation does not mention it. Alongside the link for making HTTP requests, there should be documentation about how to receive HTTP requests.

Note: if documentation is provided elsewhere, and I've just missed it - it should be duplicated or moved to this section.

Potential candidates:
- [From the same folder as the lesson on http requests](https://github.com/nodejs/nodejs.dev/blob/aa4239e87a5adc992fdb709c20aebb5f6da77f86/content/learn/node-js-web-server/node-http-server.en.md) 
- [From w3 schools tutorial](https://www.w3schools.com/nodejs/nodejs_http.asp)  
  - (the W3 schools tutorial is "optional extra credit", therefore cannot be pointed to as the source of this documentation.)
- [From nodejs](https://nodejs.org/en/docs/guides/anatomy-of-an-http-transaction)

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/nodejs-getting-started

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Discord Name

bobslim

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by mekbibDev**
## Because

There is currently no explanation as to what http.createserver does except for a snippet example. So I've added this to explain it a little better

## This PR

- Adds a reading assignment
- Gives a small summary on the same line as the reading assignment.

## Issue

Closes #26558

## Additional Information

- I tried linking it with a # relative link but it just never worked. 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by mekbibDev** [comment]
Thanks for the feedback. I'll make the chagnes. however i've tried using the anchor thingy before and i've tried the one that you have requested as well but it doesn't anchor correctly. it anchors between message.between or message.connection. is it different for you?

File: nodeJS/introduction_to_nodeJS/getting_started.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ By the end of this lesson, you should be able to do the following:
     - Learn quickly about .env files and how we use them [here](https://nodejs.org/en/learn/command-line/how-to-read-environment-variables-from-nodejs)! This will become very important in the future when working with databases and other sensitive credentials!
   - HTTP Module
     - Learn [how to make HTTP requests with Node](https://github.com/nodejs/nodejs.dev/blob/aa4239e87a5adc992fdb709c20aebb5f6da77f86/content/learn/node-js-web-server/node-make-http-requests.en.md).
+    - Checkout createserver from the [http module](https://nodejs.org/api/http.html). The documentation shows optional parameters it can accept which you may or may not use. For now, you just need to know that createserver helps us create an HTTP server that accepts handlers that will be executed every time we get a request.
```
[the remark this answers is not in the document]
Comment (mekbibDev) [reply nodeJS/introduction_to_nodeJS/getting_started.md:37]: https://github.com/TheOdinProject/curriculum/pull/27080#issuecomment-1893842652

**Comment by fortypercenttitanium** [comment]
> Thanks for the feedback. I'll make the chagnes. however i've tried using the anchor thingy before and i've tried the one that you have requested as well but it doesn't anchor correctly. it anchors between message.between or message.connection. is it different for you?

Whoa, that is a weird bug. If you use the URL while already on the page, it scrolls correctly, but if you navigate their from a different page, it scrolls to the wrong place.

In this case, just remove that part of my suggestion and we'll get it merged.

**Review by fortypercenttitanium — APPROVED** [review]

**Review by fortypercenttitanium — CHANGES_REQUESTED** [review]
Okay, so we have some linting issues. Let's add the "Lesson overview" heading above the second paragraph in the Introduction section. Also, there is an extra trailing space under line 52, let's get rid of it. Thanks.

**Comment by mekbibDev** [comment]
oh no another one. i think i now the problem with this one tho

**Review by fortypercenttitanium — APPROVED** [review]

File: nodeJS/introduction_to_nodeJS/getting_started.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ Like we learned in the introduction lesson, Node.js is really just JavaScript. S
 
 This lesson will take you through a tutorial that will teach you the basic modules and functions that you need to get up and running with Node.js. The project that comes at the end of this section will ask you to use Node to create a basic website that will include an `Index`, `About` and `Contact Me` page. So while learning the topics in this lesson, be on the lookout for things that might help you complete the project.
 
+### Lesson overview
```
Comment (fortypercenttitanium) [inline nodeJS/introduction_to_nodeJS/getting_started.md:7]: Not here, before the above paragraph

File: nodeJS/introduction_to_nodeJS/getting_started.md [hunk 3]
```diff
@@ -4,6 +4,8 @@ Like we learned in the introduction lesson, Node.js is really just JavaScript. S
```
Comment (fortypercenttitanium) [inline nodeJS/introduction_to_nodeJS/getting_started.md:4]:
```suggestion
### Lesson overview

```

**Review by fortypercenttitanium — CHANGES_REQUESTED** [review]
Sorry for the accidental approval. Just one more minor change.

**Comment by mekbibDev** [comment]
oh my bad. i hadn't re read it to realise that the there was enough content there that could describe the lesson overview and  thought there was going to be a new task to add that part and for now was enough to add the lesson overview heading

**Review by fortypercenttitanium — APPROVED** [review]

**Pull request merged**